src/Scripting/CSharp/Hosting/ObjectFormatter/CSharpObjectFormatter.cs

```diff
@@ -1,267 +1,42 @@
-                    {
-                        if (arrayOpt.GetLowerBound(i) > 0)
-                        {
-                            anyNonzeroLowerBound = true;
-                            break;
-                        }
-                    }
-
-                    for (int i = 0; i < rank; i++)
-                    {
-                        int lowerBound = arrayOpt.GetLowerBound(i);
-                        int length = arrayOpt.GetLength(i);
-
-                        if (i > 0)
-                        {
-                            sb.Append(", ");
-                        }
-
-                        if (anyNonzeroLowerBound)
-                        {
-                            AppendArrayBound(sb, lowerBound, options.UseHexadecimalNumbers);
-                            sb.Append("..");
-                            AppendArrayBound(sb, length + lowerBound, options.UseHexadecimalNumbers);
-                        }
-                        else
-                        {
-                            AppendArrayBound(sb, length, options.UseHexadecimalNumbers);
-                        }
-                    }
-
-                    sb.Append(']');
-                    arrayOpt = null;
-                }
-                else
-                {
-                    AppendArrayRank(sb, type);
-                }
-
-                type = type.GetElementType();
-            }
-            while (type.IsArray);
-
-            return sb.ToString();
-        }
-
-        private void AppendArrayBound(StringBuilder sb, long bound, bool useHexadecimalNumbers)
-        {
-            if (bound <= int.MaxValue)
-            {
-                sb.Append(FormatLiteral((int)bound, useHexadecimalNumbers));
-            }
-            else
-            {
-                sb.Append(FormatLiteral(bound, useHexadecimalNumbers));
-            }
-        }
-
-        private static void AppendArrayRank(StringBuilder sb, Type arrayType)
+        protected override bool IsHiddenMember(MemberInfo member)
         {
-            sb.Append('[');
-            int rank = arrayType.GetArrayRank();
-            if (rank > 1)
-            {
-                sb.Append(',', rank - 1);
-            }
-            sb.Append(']');
+            // Generated fields, e.g. "<property_name>k__BackingField"
+            return GeneratedNames.IsGeneratedMemberName(member.Name);
         }
 
-        internal override string FormatMemberName(System.Reflection.MemberInfo member)
+        protected override string FormatRefKind(ParameterInfo parameter)
         {
-            return member.Name;
+            return parameter.IsOut
+                ? parameter.IsIn
+                    ? "ref"
+                    : "out"
+                : "";
         }
 
-        internal override bool IsHiddenMember(System.Reflection.MemberInfo member)
+        protected override bool TryFormatCompositeObject(object obj, out string value, out bool suppressMembers)
```

Comment:
Per offline discussion, I think we can probably cut this for now (until we can provide a complete mechanism for formatting recursive members).